app/shared_features/set_card/analysis/SetAnalysis.js

```diff
@@ -0,0 +1,261 @@
+                    case 'Last Rep':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getLastPKH(this.props.set), quantifier, bigMetric);
+                    case 'Minimum':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getMinPKH(this.props.set), quantifier, bigMetric);
+                    case 'Maximum':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getMaxPKH(this.props.set), quantifier, bigMetric);
+                }
+            case 'ROM':
+                switch(quantifier) {
+                    case 'Average':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getAvgROM(this.props.set), quantifier, bigMetric);
+                    case 'Abs Loss':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getAbsLossOfROMs(this.props.set), quantifier, bigMetric);
+                    case 'First Rep':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getFirstROM(this.props.set), quantifier, bigMetric);
+                    case 'Last Rep':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getLastROM(this.props.set), quantifier, bigMetric);
+                    case 'Minimum':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getMinROM(this.props.set), quantifier, bigMetric);
+                    case 'Maximum':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getMaxROM(this.props.set), quantifier, bigMetric);
+                }
+            case 'Dur': 
+                switch(quantifier) {
+                    case 'Average':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getAvgDuration(this.props.set), quantifier, bigMetric);
+                    case 'Abs Loss':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getAbsLossOfDurations(this.props.set), quantifier, bigMetric);
+                    case 'First Rep':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getFirstDuration(this.props.set), quantifier, bigMetric);
+                    case 'Last Rep':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getLastDuration(this.props.set), quantifier, bigMetric);
+                    case 'Minimum':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getMinDuration(this.props.set), quantifier, bigMetric);
+                    case 'Maximum':
+                        return this._renderAnalysis(metric, CollapsedMetrics.getMaxDuration(this.props.set), quantifier, bigMetric);
+                    case 'Best Ever':
+                        return this._renderAnalysis(metric, SetsSelectors.getBestDurationEver(this.props.state, this.props.set), quantifier, bigMetric);
```

Comment:
- [ ] Ah this is why you passed in state. Don't do this here, have the actual values you need displayed be set in the CONTAINER rather than pull it out of the set in the view. You're giving the View too much logic duties, and that's why you had to do something crazy like pass in the entire fucking state. That's a red flag anytime you work in redux